pep-0517.txt

```diff
@@ -218,6 +218,23 @@ metadata. The directory passed in by the build frontend MUST be
 identical to the directory created by ``get_wheel_metadata``,
 including any unrecognized files it created.
 
+::
+
+  def prepare_build_files(build_directory, config_settings):
+      ...
+
+Must copy or create any files needed to build this package into
+``build_directory``. For instance, it is expected  that ``pyproject.toml`` will
+exist in the root of this directory after this hook runs. For tools such as
+`setuptools_scm <https://github.com/pypa/setuptools_scm>`_, this may include
+extracting some information from a version control system.
+The ``build_wheel`` hook will subsequently be run from the ``build_directory``
+populated by this hook.
+
+Optional. If this hook is not defined, the copying step will be skipped, and the
+``build_wheel`` hook invoked in the original source directory. In this case, the
+backend is responsible for not cluttering up the source directory.
```

Comment:
I don't feel strongly about this, my rationale is just that if the backend knows it can build a wheel without creating clutter in the source directory (as flit does), then we can save copying the files. But the normal operation should be installing from a pre-built wheel anyway, so I don't think performance is that critical for this step.